### ISSUE TITLE
Problems with Padatious

### ISSUE THREAD

**Issue by aaronchantrill** [NaomiProject/Naomi#306]
## Description

The Padatious TTI plugin is returning wildly incorrect results right now.

## Expected Behavior

Saying "Naomi, play music" should result in Naomi starting the music mode, not Naomi shutting down.

## Actual Behavior

Naomi seems to react randomly and sometimes even return a completely empty result.

## Possible Fix

I'm going to convert my [Intent_parser_tests](https://github.com/aaronchantrill/intent_parser_tests) scripts into unit tests for the three current TTI plugins. This should help me track down the issue.

**Comment by aaronchantrill** [comment]
Actually, it looks like this is because of some changes to the Padatious engine itself. It appears that the Mycroft folk have made it more precise, which is causing a lot of my requests to fail.

I am expecting fuzzy matches to work since STT is so imprecise, and because it is difficult for a plugin author to anticipate every way the user may activate the plugin.

It appears that including the wake word in the phrase (which happens a lot when passive_listen is activated) when it was not included in the training templates is also causing a lot of trouble, which is easy to take care of in those cases where the wake word appears in the active transcript, but more difficult when the wake word itself is misheard by the active listener.

Also, queries like "play {artist}" are no longer returning fuzzy matches. It used to be that I could define the "artist" keyword as a list of specific artists but Padatious would also return a match of an artist not on the list. This is causing Padatious to flail.

I was trying to test the MPD plugin. I loaded the Anais Mitchell Hadestown album and created a playlist called "hadestown" which is added to the MPDControl_Playlist keyword list, and the templates are defined as ['PLAY SOMETHING', 'PLAY MUSIC', 'PLAY {MPDControl_Playlist}'. Here are the responses I was getting when asking "Naomi, play Hadestown":

```
TELL ME PLAY HADESTOWN - ClockIntent - 0.21
NAOMI PLAY HADESTOWN - ShutdownIntent - 0.14
THE IN PLAY HADESTOWN - ShutdownIntent - 0.13
HADESTOWN - MeaningOfLifeIntent - 0.09
PLAY HADESTOWN - MPDControlIntent - 1.0
```

So an exact match with the "PLAY {MPDControl_Playlist}" template gets 100%, but it's odd that including some extra words generates such wildly different results. The word "TELL" appears in the "ChuckIntent" and "JokeIntent" in addition to the ClockIntent, but the word "PLAY" only appears in the MPDControlIntent. The ShutdownIntent templates only include "SHUTDOWN" and "TURN YOURSELF OFF" so why it would be activated by the words "NAOMI" or "THE IN" is beyond me.

Right now there are 25 versions of Padatious on PyPi for the ARM architecture going back to version 0.1.1. I'm currently running 0.4.8 (the latest version). I'm going to try installing different versions and see if I can determine the best version to use with Naomi.

**Comment by aaronchantrill** [comment]
I checked a bunch of versions. The best is probably version 0.4.5 which was able to identify the correct intent in all but the most extreme cases (it categorized "PLAY THE WHO" as GameIntent and "WHAT'S HAPPENING TOMORROW" as WeatherIntent, both of which are pretty understandable) but still failed to pick up on keyword substitution a lot of the time, unless the wording exactly matches the template (even then, when given both a list and non-list keyword in the same template it would only report the non-list keyword).

In my original intent tests, I specifically picked examples that matched vaguely rather than exactly. I have replaced all the tests with exact matches in my unittest test cases, because if padatious gets anything wrong, the whole test fails.

I have to say, I'm pretty disappointed in Padatious. I am curious how Adapt and Padatious interface within the Mycroft intent parser.
